internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -793,6 +850,7 @@ func resourceLinuxVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta int
 	}
 
 	d.Set("do_not_run_extensions_on_overprovisioned_machines", props.DoNotRunExtensionsOnOverprovisionedVMs)
+	d.Set("host_group_id", props.HostGroup.ID)
```

Comment:
This is throwing nil pointer during the test, need to check nil of `props.HostGroup` and `props.HostGroup.ID`

Reply:
@myc2h6o thanks for the heads up, this is now fixed. 🚀